src/Interpreters/InterpreterInsertQuery.cpp

```diff
@@ -285,7 +285,7 @@ Chain InterpreterInsertQuery::buildChainImpl(
 
     /// Do not squash blocks if it is a sync INSERT into Distributed, since it lead to double bufferization on client and server side.
     /// Client-side bufferization might cause excessive timeouts (especially in case of big blocks).
-    if (!(settings.insert_distributed_sync && table->isRemote()) && !async_insert && !no_squash && !(query && query->watch))
+    if (!(settings.insert_distributed_sync && table->isRemote(context_ptr)) && !async_insert && !no_squash && !(query && query->watch))
```

Comment:
According to the comment, this is only relevant to insert into distributed. Not relevant for parallel replicas.
(passing context to isRemote method unacceptable).